Greedy query planning for large connected components

joka921 opened the pull request:
So far, the query planning took very long when the query graph has a large connected component. With this change, the query planner counts the number of connected subgraphs for each connected component. If that number exceeds a certain budget, the query planning for the component is done using so-called Greedy Operator Ordering (GOO). For a connected component with n nodes, GOO considers a quadratic number of plans, whereas the optimal dynamic programming potentially considers an exponential number of plans. The budget can be controlled via the runtime parameter `query-planning-budget` (default: 1500).

hannahbast reviewed (changes requested):
Had a first look with Johannes. Awesome that this was possible with so little code. Speaks for the quality of the question planner code!

hannahbast reviewed (changes requested):
Thorough 1-1 with Johannes, during which I made various changes that I will commit next

hannahbast commented:
@joka921 Thanks a lot + looks great now. One more nitpick: `expectDifferentPlanners` looks like a misnomer (look at its many applications in `test/QueryPlannerTest.cpp`). Can you find a better name?

joka921 commented:
@hannahbast 
1. The default function (check with both planners) is now again called only `expect` (reduce the change), The other ones have explicit names that state, that you only want greedy or dynamic programming respectively.
2. I went with `query-planning-budget` which seems to be consistent with the current commit message.
3. Agreed and dropped
4. I replaced this `if` by an assertion that explicitly tells us where we have to change things should we ever lift the restriction of 64 elements.

The other suggestions are from Codecov etc.
